packages/jest-snapshot/src/State.js

```diff
@@ -12,72 +12,155 @@
 
 import type {Path} from 'types/Config';
 
-const {forFile} = require('./SnapshotFile');
-import type {SnapshotFile} from './SnapshotFile';
+const {
+  saveSnapshotFile,
+  fileExists,
+  getSnapshotData,
+  getSnapshotPath,
+  keyToTestName,
+  serialize,
+  testNameToKey,
+  unescape,
+} = require('./utils');
+const fs = require('fs');
 
 class SnapshotState {
   _counters: Map<string, number>;
   _index: number;
+  _snapshotPath: Path;
+  _dirty: boolean;
+  _snapshotData: {[key: string]: string};
+  _uncheckedKeys: Set<string>;
   added: number;
+  failedTests: Set<string>;
   matched: number;
-  snapshot: SnapshotFile;
+  skippedTests: Set<string>;
   unmatched: number;
   update: boolean;
   updated: number;
 
-  constructor(testPath: Path, update: boolean) {
+  constructor(
+    testPath: Path,
+    update: boolean,
+    snapshotPath?: string,
```

Comment:
awesome!
we'll probably change the API again a little bit before the release (we'll need to add interactive update), but i'll make sure we keep those DIs there